Tella/Data/Database/Vault/VaultDataSource.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Copyright © 2023 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+
+class VaultDataSource : VaultDataSourceInterface {
+    
+    
+    var database : VaultDatabase
+
+    init(key: String?) {
+        self.database = VaultDatabase(key: key)
+    }
+
```

Comment:
Consider using dependency injection for the VaultDatabase instance instead of creating it directly in the initializer. 

 ```
 var database: VaultDatabaseProtocol
    init(database: VaultDatabaseProtocol) {
        self.database = database
    }
```